### ISSUE TITLE
Eclipse Che Installation in AWS EC2, failure to start a workspace

### ISSUE THREAD

**Issue by svkr2k** [eclipse-che/che#13690]
### Description
Trying to run `che-server:7.0.0-RC-2.0`. che-server installs and runs fine, able to view dashboard and stacks/devfiles.

But, when i try to create a workspace, it fails. The logs are given below.

Some of the experts in a forum feel that the che master pod is unable to communicate with the theia pod.

In local PC everyting works fine and able to create workspaces.
But, in AWS EC2, dashboard comes up, but creating workspace fails.

### Reproduction Steps

```bash
minikube start --cpus 2 --memory 4096 --extra-config=apiserver.authorization-mode=RBAC --vm-driver=none
kubectl create clusterrolebinding add-on-cluster-admin --clusterrole=cluster-admin --serviceaccount=kube-system:default
kubectl create serviceaccount tiller --namespace kube-system
kubectl apply -f ./tiller-rbac.yaml
helm init --service-account tiller --wait
minikube addons enable ingress
helm dependencies update --skip-refresh ./
minikube ip
export CHE_DOMAIN=$(minikube ip)
helm upgrade --install che --force --namespace che --set global.ingressDomain=${CHE_DOMAIN}.nip.io --set cheImage=eclipse/che-server:7.0.0-RC-2.0 ./
```

**OS and version:**
Ubuntu 18.04
AWS EC2 instance

**Diagnostics:** 
Here is the log:

```
2019-07-02 13:34:48,148[nio-8080-exec-5]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 433]   - Starting workspace 'che/wksp-0m6t' with id 'workspaceh47e7hjup6tqzgqd' by user 'che'
2019-07-02 13:35:22,846[nio-8080-exec-2]  [WARN ] [a.c.w.i.BasicWebSocketEndpoint 145]  - Web socket session error
2019-07-02 13:35:22,847[nio-8080-exec-2]  [WARN ] [a.c.w.i.BasicWebSocketEndpoint 129]  - Closing unidentified session`
2019-07-02 13:39:18,077[aceSharedPool-1]  [WARN ] [.i.k.KubernetesInternalRuntime 245]  - Failed to start Kubernetes runtime of workspace workspaceh47e7hjup6tqzgqd. Cause: Server 'theia' in machine 'theia-ide3dy' not available.
2019-07-02 13:39:18,834[aceSharedPool-1]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 856]   - Workspace 'che:wksp-0m6t' with id 'workspaceh47e7hjup6tqzgqd' start failed
```

**Comment by tsmaeder** [comment]
@svkr2k how are you creating and starting the workspace?

**Comment by benoitf** [comment]
I think it's a duplicate of https://github.com/eclipse/che/issues/13435

**Comment by tsmaeder** [comment]
@benoitf Looks likey, yes. The logs here and on the other bug look uninteresting. What additional info could we request that would help us pin down the issue?

**Comment by tsmaeder** [comment]
Tentatively marking this as 7.0.0, not starting is not good.

**Comment by l0rd** [comment]
I suspect that `minikube ip` doesn't provide the good IP address in the case of EC2. I mean it provides the internal IP but not the public one. See [EC2 user guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-instance-addressing.html#using-instance-addressing-common) for more details.

@skabashnyuk @sleshchenko do you think that this may be the problem? Do you have other ideas about why wsmaster and theia cannot communicate?

**Comment by svkr2k** [comment]
> @svkr2k how are you creating and starting the workspace?

@tsmaeder , thank you for your question.
I was able to open http://che-che.<ipaddr>.nip.io and see the che dashboard.
In the dashboard, i selected a stack from the list and clicked on Create workspace button to create the workspace.

Hope this helps. Thank you for looking into this. I would really be happy if i get a solution for the issue.

**Comment by skabashnyuk** [comment]
@svkr2k just to clarify your environment. Are you running Che locally with minikube or on AWS?

**Comment by andy316x** [comment]
I experience this issue if running in multi-user mode, interestingly if I deploy Che in single-user mode Theia loads successfully, however it is then not able to create terminals. I am also using Che in AWS.

**Comment by benoitf** [comment]
@andy316x with which browser for terminals ? There is an issue with firefox

**Comment by andy316x** [comment]
@benoitf I am using Chrome so not sure if that is the problem

**Comment by svkr2k** [comment]
@skabashnyuk , 

> @svkr2k just to clarify your environment. Are you running Che locally with minikube or on AWS?

I'm, running che on AWS in single-user mode and using any of the already available stack/devfile from the dashboard.

(Kindly note that when i run che in locally in ubuntu or windows PC, it works fine.)

**Comment by skabashnyuk** [comment]
@svkr2k I have no AWS EC2 environment available to test. Can you try to install che with https://github.com/che-incubator/chectl ?

**Comment by svkr2k** [comment]
> @svkr2k just to clarify your environment. Are you running Che locally with minikube or on AWS?

Hi @skabashnyuk , to clarify further: I'm running Che on AWS "with minikube".

**Comment by svkr2k** [comment]
Hi all,
I have a question regarding the above issue:
While creating a workspace from dashboard, will the che-host try to communicate with workspace container using .nip.io url?

**Comment by skabashnyuk** [comment]
@svkr2k can you try the way to install che how @l0rd suggested here? https://github.com/eclipse/che/issues/13838#issuecomment-511723339

**Comment by svkr2k** [comment]
@skabashnyuk , sure, i shall try that : https://github.com/eclipse/che/issues/13838#issuecomment-511723339

Also, please note that in AWS EC2 instance, i start minikube using the following:
`--vm-driver=none`

**Comment by svkr2k** [comment]
> @svkr2k can you try the way to install che how @l0rd suggested here? [#13838 (comment)](https://github.com/eclipse/che/issues/13838#issuecomment-511723339)

Thank you very much for providing inputs.

when i tried chectl server:start as shown in the above link:
`~$ sudo chectl server:start --installer=helm --multiuser --platform=k8s --tls --self-signed-cert --domain=10.9.2.247.nip.io --cheimage=eclipse/che-server:7.0.0-rc-3.0
sudo: unable to resolve host ip-10-9-2-247
  ✔ ✈️  Kubernetes preflight checklist
    ✔ Verify if kubectl is installed
    ✔ Verify remote kubernetes status...done.
    ✔ Verify domain is set...set to 10.9.2.247.nip.io.
  ❯ 🏃‍  Running Helm to install Che
    ✔ Verify if helm is installed
    ✖ Check for TLS prerequisites
      → TLS option is enabled but che-tls secret does not exist in default namespace. Example on how to create the secret: kubectl create secret generic che-tls --from-literal=ACME_EMAIL=my@email-address.com
      Create Tiller Role Binding
      Create Tiller Service Account
      Create Tiller RBAC
      Create Tiller Service
      Preparing Che Helm Chart
      Updating Helm Chart dependencies
      Deploying Che Helm Chart
Error: TLS option is enabled but che-tls secret does not exist in default namespace. Example on how to create the secret: kubectl create secret generic che-tls --from-literal=ACME_EMAIL=my@email-address.com
    at HelmHelper.<anonymous> (/snapshot/chectl/lib/installers/helm.js:0:0)
    at Generator.next (<anonymous>)
    at fulfilled (/snapshot/chectl/node_modules/tslib/tslib.js:107:62)`

Sorry, i havent used TLS, could you please guide next steps... Thank you very much !

**Comment by benoitf** [comment]
$ kubectl create namespace cert-manager
$ kubectl label namespace cert-manager certmanager.k8s.io/disable-validation=true
$ kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.8.1/cert-manager.yaml --validate=false

**Comment by svkr2k** [comment]
Greetings, @benoitf . Thank you !

Here are the steps i followed:
```
$ sudo minikube start --memory=4096 --vm-driver=none
$ sudo kubectl create secret generic che-tls --from-literal=ACME_EMAIL=my_id@gmail.com

$ kubectl create namespace cert-manager
$ kubectl label namespace cert-manager certmanager.k8s.io/disable-validation=true
$ kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.8.1/cert-manager.yaml --validate=false

$ sudo chectl server:start --installer=helm --multiuser --platform=k8s --tls --self-signed-cert --domain=10.9.2.247.nip.io --cheimage=eclipse/che-server:7.0.0-rc-3.0
```
Here are the other errors that i could see:
```
~$ sudo chectl server:start --installer=helm --multiuser --platform=k8s --tls --self-signed-cert --domain=10.9.2.247.nip.io --cheimage=eclipse/che-server:7.0.0-rc-3.0
sudo: unable to resolve host ip-10-9-2-247
  ✔ ✈️  Kubernetes preflight checklist
    ✔ Verify if kubectl is installed
    ✔ Verify remote kubernetes status...done.
    ✔ Verify domain is set...set to 10.9.2.247.nip.io.
  ✔ 🏃‍  Running Helm to install Che
    ✔ Verify if helm is installed
    ✔ Check for TLS prerequisites che-tls secret exist.
    ✔ Create Tiller Role Binding...it already exist.
    ✔ Create Tiller Service Account...it already exist.
    ✔ Create Tiller RBAC
    ✔ Create Tiller Service...it already exist.
    ✔ Preparing Che Helm Chart...done.
    ✔ Updating Helm Chart dependencies...done.
    ✔ Deploying Che Helm Chart...done.
  ❯ ✅  Post installation checklist
    ❯ PostgreSQL pod bootstrap
      ✖ scheduling
        → ERR_TIMEOUT: Timeout set to pod wait timeout 300000. podExist: false, currentPhase: undefined
        downloading images
        starting
      Keycloak pod bootstrap
      Che pod bootstrap
      Retrieving Che Server URL
      Che status check
Error: ERR_TIMEOUT: Timeout set to pod wait timeout 300000. podExist: false, currentPhase: undefined
    at KubeHelper.<anonymous> (/snapshot/chectl/lib/api/kube.js:0:0)
    at Generator.next (<anonymous>)
    at fulfilled (/snapshot/chectl/node_modules/tslib/tslib.js:107:62)
```

**Comment by svkr2k** [comment]
Hi @benoitf , @skabashnyuk .
I believe that the urls used for communicating with the containers could be based on .nip.io.
As an alternative, could you kindly send me a link to the documentation or steps to setup che to use domain name based urls instead of <ip addredd>.nip.io ?
Something similar to "http://che-che.myapp.mydomain.com"?
Kindly let me know how to setup like that, so that i can test that too... Thank you for your help.

**Comment by skabashnyuk** [comment]
@svkr2k I believe that this is ```-b``` parameter of chectl  https://www.eclipse.org/che/docs/che-7/che-quick-starts.html

**Comment by skabashnyuk** [comment]
@benoitf when you made your test on AWS did you use minikube aswell?

**Comment by l0rd** [comment]
@svkr2k when you first opened this issue you tried to install [single user Che with TLS disable](https://github.com/eclipse/che/issues/13690#issue-464017986)d. At some point though you tested [with TLS and multiuser](https://github.com/eclipse/che/issues/13690#issuecomment-512063268). Have you been able to solve your original problem (Che single user without tls on AWS)? 

I am asking because we have some [open](#13869) [issues](#12972) about deploying Che with TLS using helm charts (hence this would be a duplicate) and I would close this one if it your original issue was fixed.

**Comment by l0rd** [comment]
@rhopp @slemeur @tsmaeder @nickboldt I have removed the priority and the milestone here because we don't know yet if that's a duplicate, an issue that has been solved or a brand new one.

**Comment by svkr2k** [comment]
Hi @l0rd , the original issue was due to the .nip.io urls being blocked in my domain. Now, i'm able to create a workspace as we have 'unblocked' nip.io based url.
But, my goal is to use hostname-based url for che. we could achieve this using defaulthost.yaml while running the helm upgrade.

**Comment by l0rd** [comment]
@svkr2k what you are describing is an existing issue about default-host/single-host: https://github.com/eclipse/che/issues/12971. Unfortunately that's something that we haven't solved yet. We will need a few weeks to fix it properly.

But, if you multi-host is an option (i.e. you are ok to use a wildcard SSL certificate) that's possible to configure it [with a workaround](https://github.com/eclipse/che/issues/13869#issuecomment-513751078). We have an ongoing PR to avoid the workaround and make it easier to deploy Che on AWS and GCP: https://github.com/eclipse/che/issues/12971. 

And in the meantime we are writing the documentation as well but we are kind of blocked by the issues above.

**Comment by mshaposhnik** [comment]
So I'm closing current issue since it will be fixed by #12971
